spek-table-driven-extension/src/main/kotlin/org/jetbrains/spek/table/TableDriven.kt

```diff
@@ -0,0 +1,320 @@
+package org.jetbrains.spek.table
+
+import org.jetbrains.spek.api.dsl.Pending
+import org.jetbrains.spek.api.dsl.Spec
+import org.jetbrains.spek.api.dsl.TestBody
+import org.jetbrains.spek.api.dsl.TestContainer
+
+data class Example1<out P1> internal constructor(val p1: P1)
+
+fun <P1> example(p1: P1): Example1<P1> = Example1(p1)
+fun <P1> Spec.with(vararg examples: Example1<P1>, testContent: TestContainer. (P1) -> Unit) {
+    examples.forEach {
+        val proxyContainer = object : TestContainer by this@with {
```

Comment:
Thanks !
Looking at the SpecBody, I see the `memoized` and  `before/afterXXX` functions. I can't see a use case to that with table driven tests, do you ?
If not, I think introducing a new interface that contains only group and action would be nice. What do you think ?